exercises/two-bucket/two-bucket.example.ts

```diff
@@ -0,0 +1,91 @@
+export default class TwoBucket {
+    starter: string
+    x: number
+    y: number
+    z: number
+    goalBucket: string
+    otherBucket: number
+
+    constructor(x: number, y: number, z: number, starter: string) {
+      this.starter = starter
+      this.x = x
+      this.y = y
+      this.z = z
+    }
+    reachedGoal(measurements: number[]): boolean {
+      if (measurements[0] === this.z || measurements[1] === this.z) {
+        if (measurements[0] === this.z) {
+          this.goalBucket = 'one'
+          this.otherBucket = measurements[1]
+        } else {
+          this.goalBucket = 'two'
+          this.otherBucket = measurements[0]
+        }
+        return true
+      }
+      return false
+    }
+    bigFirst(measurements: number[], moveCount: number, prBool: boolean): number {
+      let j = measurements[0],
+        k = measurements[1]
+      while (!this.reachedGoal(measurements)) {
+        if (k > this.x && j === 0 && moveCount === 0) {
+          j = this.x
+          k = this.y - j
+        } else if (j === this.x) {
+          j = 0
+        } else if (k > this.x && j !== 0 || k > this.x && prBool) {
+          k -= (this.x - j)
+          j = this.x
+        } else if (k > this.x || j === 0) {
+          j = k
+          k -= j
+        } else if (k === 0) {
+          k = this.y
+        }
+        measurements = [j, k]
+        moveCount++
+        prBool = !prBool
+      }
+      return moveCount
+    }
+    smallFirst(measurements: number[], moveCount: number, prBool: boolean): number {
+      let j = measurements[0],
+        k = measurements[1]
+      while (!this.reachedGoal(measurements)) {
+        if (j === this.x && moveCount === 0) {
+          j = 0
+          k = this.x
+        } else if (j === 0) {
+          j = this.x
+        } else if (j === this.x && k < this.y) {
+          const tempK = k
+          k + j > this.y ? k = this.y : k = tempK + j
+          tempK + j > this.y ? j -= (this.y - tempK) : j = 0
+        } else if (k === this.y) {
+          k = 0
+        } else if (k === 0 && j < this.x) {
+          k = j
+          j = 0
+        }
+        measurements = [j, k]
+        moveCount++
+        prBool = !prBool
+      }
+      return moveCount
+    }
+    moves() {
+      let j = 0,
+        k = 0 // j will be running val of bucket one, k = running val of bucket two
+      this.starter === 'one' ? j = this.x : k = this.y
```

Comment:
starter should be an enum. 
https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-4.html

Reply:
I update the track to use the latest TS which has support to string enums.